### DISCUSSION TITLE
Model transfer between GeometryService and PyPrimeMesh

### DISCUSSION THREAD

**Discussion by dkowollik** [ansys/pyansys-geometry#1381] [Q&A] [answered]
Hi @RobPasMue,

is it possible to directly transfer models in memory between GeometryService and PyPrimeMesh? If not, what is the best (fastest) model transfer method or file type between those two ansys python packages?

**Comment by RobPasMue** [comment, marked as answer]
Hi @dkowollik!

Currently there is no option to transfer a memory in model, though internally we are working in ways to do this across the entire ecosystem with common data models. This might take some time to happen though but it is one of the working lines for the ecosystem.

The fastest way via file export/import would be to use the ".fmd" format. PyPrimeMesh has native support for that file format so my recommendation would be the following:

```python

from ansys.geometry.core import launch_modeler_with_geometry_service

modeler = launch_modeler_with_geometry_service()

design = modeler.create_design("MyDesign")

# Do your modelling operations
# ...

# Export to FMD
design.export_to_fmd()
```

The previous call will generate a file called ``MyDesign.fmd`` in your current working directory. The API ``export_to_fmd(...)`` also accepts passing in a specific directory to export the file to. See https://geometry.docs.pyansys.com/version/stable/examples/03_modeling/export_design.html#Export-to-FMD-format

PyPrimeMesh should be able to read it natively then. Something like this should work as far as I know:

```python
import ansys.meshing.prime as prime

with prime.launch_prime() as session:
    model = session.model
    block_model = "MyDesign.fmd"
    with prime.FileIO(model) as io:
       _ = io.import_cad(block_model, params=prime.ImportCADParams(model))
       print(model)
````

**Reply by RobPasMue** [reply]
FYI - PyPrimeMesh also supports SCDOCX as well although it is not native. Other native formats are parasolid text and parasolid binary - which PyAnsys Geometry also exports to. See more details here: https://prime.docs.pyansys.com/version/stable/user_guide/fileio.html

**Reply by dkowollik** [reply]
Thank you. Good to hear that you are working on a direct transfer in RAM. I had to change `prime.ImportCADParams(model)` to `prime.ImportCadParams(model)`. May be one should talk to the prime team. All imports step, 3dm and fmd files take quite long for my model, about 24s. The 3dm import into GeometryService takes only 6s and the fmd export takes 4s.

**Reply by RobPasMue** [reply]
Pinging the @ansys/pyprimemesh-maintainers  team for visibility. Import of FMD files seem to be taking quite long. Can we get any improvements there?
